chromium/rules.js

```diff
@@ -405,51 +416,61 @@ RuleSets.prototype = {
    */
   potentiallyApplicableRulesets: function(host) {
     // Have we cached this result? If so, return it!
-    var cached_item = this.ruleCache.get(host);
-    if (cached_item !== undefined) {
-      log(DBUG, "Ruleset cache hit for " + host + " items:" + cached_item.length);
+    if (this.ruleCache.has(host)) {
+      let cached_item = this.ruleCache.get(host);
+      let len = cached_item && cached_item != nullIterable ? cached_item.size : 0;
+      log(DBUG, "Ruleset cache hit for " + host + " items:" + len);
       return cached_item;
+    } else {
+      log(DBUG, "Ruleset cache miss for " + host);
     }
-    log(DBUG, "Ruleset cache miss for " + host);
 
-    var results = [];
+    // Let's search!
+    let resultSet = new Set();
+
     if (this.targets.has(host)) {
       // Copy the host targets so we don't modify them.
-      results = results.concat(this.targets.get(host));
+      let results = this.targets.get(host);
+      results.map(result => resultSet.add(result));
```

Comment:
I'm a little confused by having both `results` and `resultsSet` variables. You can just do:

```javascript
if (this.targets.has(host)) {
  results = new Set(...results, this.targets.get(host));
```